File: packages/models/src/tag.ts
import {
    getModelForClass,
    prop,
    modelOptions,
    mongoose
} from '@typegoose/typegoose';

@modelOptions({ schemaOptions: { timestamps: true, collection: 'tags' } })
class Tag {
    @prop({ required: true, ref: 'User' })
    userId: mongoose.Types.ObjectId;

    @prop({ required: true })
    name: string;

    @prop({ required: true })
    color: string;

    @prop({ default: '' })
    description: string;

    @prop({ default: false })
    isDeleted: boolean;

    @prop({ default: null })
    deletedAt: Date;
}

export default getModelForClass(Tag);
export { Tag };

File: domain/tag/entities/tag.go
package entities

import (
	"naotodoserver/domain/types"
)

// Tag 标签实体
// 用于表示用户在系统中的标签记录
// 包含用户 ID、名称、描述、颜色、排序 ID等属性
type Tag struct {
	types.EntityBase
	UserId      types.UserID
	Name        string
	Description string
	Color       string
	SortId      uint16
}
